checker/check_test.go

```diff
@@ -26,11 +31,48 @@ func TestChecker(t *testing.T) {
 	tc.TestingT(t)
 }
 
-type testChecker struct{}
+type testCheckerSuite struct{}
 
-var _ = tc.Suite(&testChecker{})
+var _ = tc.Suite(&testCheckerSuite{})
 
-func (t *testChecker) TestCheckSyncConfig(c *tc.C) {
+type mockDBProvider struct {
+	db *sql.DB
+}
+
+// Apply will build BaseDB with DBConfig
+func (d *mockDBProvider) Apply(config config.DBConfig) (*conn.BaseDB, error) {
+	return conn.NewBaseDB(d.db, func() {}), nil
+}
+
+func (s *testCheckerSuite) initMockDB(c *tc.C) sqlmock.Sqlmock {
+	db, mock, err := sqlmock.New()
+	c.Assert(err, tc.IsNil)
+	conn.DefaultDBProvider = &mockDBProvider{db: db}
+	return mock
+}
+
+func ignoreExcept(itemMap map[string]struct{}) []string {
+	items := []string{
+		config.DumpPrivilegeChecking,
+		config.ReplicationPrivilegeChecking,
+		config.VersionChecking,
+		config.BinlogEnableChecking,
+		config.BinlogFormatChecking,
+		config.BinlogRowImageChecking,
+		config.TableSchemaChecking,
+		config.ShardTableSchemaChecking,
+		config.ShardAutoIncrementIDChecking,
+	}
+	ignoreCheckingItems := make([]string, 0)
```

Comment:
`make` for some capacity or no zero len?

Reply:
address in 8ac2fd6

Reply:
could use zero length and this capacity.

Reply:
Stupid me！😂